elkodon_pal/posix/src/lib.rs

```diff
@@ -80,13 +80,14 @@ pub mod posix {
     }
 
     pub(crate) unsafe fn c_string_length(value: *const crate::posix::char) -> usize {
-        for i in 0..usize::MAX {
-            if *value.add(i) as u8 == b'\0' {
-                return i;
+        const NULL_TERMINATION: crate::posix::char = 0;
+        for i in 0..isize::MAX {
+            if *value.offset(i) == NULL_TERMINATION {
```

Comment:
Please use `is_null()` -> `value.offset(i).is_null()`

Reply:
nope, that's wrong :)